clientapi/routing/admin.go

```diff
@@ -24,6 +25,86 @@ import (
 	"github.com/matrix-org/dendrite/userapi/api"
 )
 
+func AdminCreateNewToken(req *http.Request) util.JSONResponse {
+	request := struct {
+		Token       string `json:"token"`
+		UsesAllowed int32  `json:"uses_allowed"`
+		ExpiryTime  int64  `json:"expiry_time"`
+		Length      int32  `json:"length"`
+	}{}
+
+	if err := json.NewDecoder(req.Body).Decode(&request); err != nil {
+		return util.JSONResponse{
+			Code: http.StatusBadRequest,
+			JSON: spec.Unknown("Failed to decode request body: " + err.Error()),
+		}
+	}
+	token := request.Token
+	if len(token) > 0 {
+		if len(token) > 64 {
+			return util.MatrixErrorResponse(
+				http.StatusBadRequest,
+				string(spec.ErrorInvalidParam),
+				"token must not be empty and must not be longer than 64")
+		}
+		is_token_valid, _ := regexp.MatchString("^[[:ascii:][:digit:]_]*$", token)
```

Comment:
Please use `camelCase`, so `isTokenValid` instead. (Same for other variables, like `usesAllowed` instead of `uses_allowed`)